ScopeLift/endaoment
# Setup frontend

### Overview 

Contains initial frontend setup. Designed with a "factory" approach in mind, so when you go to the home page there is one "featured cause". Each thing called a "cause" is really one instance of an endaoment contract. (I'm open to alternative names if you don't like that one). 

Currently in the app you can either:

1. Click "Find Causes" in the header to view all endaoments
2. Click the featured cause to view its "Cause Details" page
3. Log in

The cause details page will (but does not currently) show details on current and past proposals as well as current and past streams. Eventually there should also be an area to create a new cause but that is not implemented yet. Also please feel free to suggest alternative copy on the home page.

### Testing

Follow the steps below to make sure the app runs

1. Create a file called `.env` with the following content:

    ```bash
    export BLOCKNATIVE_API_KEY=yourBlocknativeApiKey
    export INFURA_ID=yourInfuraId
    ```

2. Install packages with `npm install`.
3. Run the app with `npm run dev`.
